src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -187,16 +201,17 @@ public function generateMailsAction(Request $request)
      * @throws Exception
      */
     #[AdminSecurity("is_granted('update', request.get('_legacy_controller'))")]
-    public function saveConfigurationAction(Request $request)
-    {
-        /** @var FormHandlerInterface $formHandler */
-        $formHandler = $this->getMailThemeFormHandler();
+    public function saveConfigurationAction(
+        Request $request,
+        #[Autowire(service: 'prestashop.admin.mail_theme.form_handler')]
+        FormHandlerInterface $formHandler
+    ): Response {
         /** @var Form $form */
         $form = $formHandler->getForm()->handleRequest($request);
 
         if ($form->isSubmitted()) {
             if (!$form->isValid()) {
-                $this->flashErrors($this->getFormErrorsForJS($form));
+                $this->addFlashErrors($this->getFormErrorsForJS($form));
```

Comment:
It's weird that I used `getFormErrorsForJS` here because this is not returned as a JSON response, I think maybe we can replace this with:
```suggestion
                $this->addFlashFormErrors($form);
```

In the whole file because there are two weird usages, unless it messes with the UI tests in which case let's keep it that way